### PR TITLE
Fix #51 segfault

### LINKED ISSUE #51: Runtime segfaults
* https://github.com/AppImage/type2-runtime/actions/runs/10193445191 segfaults.
* https://github.com/AppImage/type2-runtime/actions/runs/10122947649 (#43)" still worked. Also, that build didn't have "fuse3" in the filename yet.

### PR DESCRIPTION
**Pull request by probonopd**
This seems to fix

* #51

@TheAssassin I am going to merge this as an emergency fix (since the runtime segfaults without this, this won't make it worse), please do carefully review afterwards nonetheless. Thanks!

### CODE REVIEW

File: src/runtime/runtime.c [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * of this software and associated documentation files (the "Software"), to deal
  * in the Software without restriction, including without limitation the rights
  * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
+ * copies of the Software, and to permit persons to whom the Softwaref is
```
Comment (stsp) [inline src/runtime/runtime.c]: Really?

File: src/runtime/runtime.c [hunk 2]
```diff
@@ -16,7 +16,7 @@
  * of this software and associated documentation files (the "Software"), to deal
  * in the Software without restriction, including without limitation the rights
  * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Softwaref is
+ * copies of the Software, and to permit persons to whom the Software is
```
Comment (stsp) [inline src/runtime/runtime.c]: Ah, ok. :)

File: src/runtime/runtime.c [hunk 3]
```diff
@@ -862,25 +862,26 @@ bool rm_recursive(const char* const path) {
...
 }
 
-void build_mount_point(char* mount_dir, const char* const argv0, char const* const temp_base, const size_t templen) {
+void build_mount_point(char* mount_dir, const char* const argv0, const char* const temp_base, const size_t templen) {
     const size_t maxnamelen = 6;
+    const size_t prefix_len = 8; // Length of "/.mount_"
+    const size_t suffix_len = 6;  // Length of "XXXXXX"
+
+    // Create a modifiable copy of argv0
+    char argv0_copy[PATH_MAX]; // Ensure this is large enough for your use case
```
Comment (TheAssassin) [inline src/runtime/runtime.c:871]: ***Please stop using `PATH_MAX`!*** Not only me but others have told you more than once that this variable's name is really misleading. One can allocate the buffer to the correct size directly in this case. (Or just use C++...)

Comment (TheAssassin) [reply src/runtime/runtime.c:871]: See, e.g., https://github.com/AppImage/type2-runtime/pull/38, which attempts to fix some of this mess...

Comment (probonopd) [reply src/runtime/runtime.c:871]: Point taken, but at least with this change we don't get an immediate segfault like we had before.

Comment (TheAssassin) [reply src/runtime/runtime.c:871]: The runtime really needs way more love to fix some of those C-isms. I wish C had a real string concept. But then again, we wanted to rewrite the runtime in another language anyway. The main issue with that is that libfuse is not available for languages like Rust, at least nothing that would be maintained or remotely up to date. Languages like Go don't fix the real issues we have either. C++ would fix most of the complaints but including the STL causes the binary to grow size wise.

Comment (probonopd) [reply src/runtime/runtime.c:871]: Indeed. So C it is. Just needs some love.

Comment (TheAssassin) [reply src/runtime/runtime.c:871]: Well, I have some hope for Rust using FFI. That'd still eliminate most of the issues on our side.

Comment (stsp) [reply src/runtime/runtime.c:871]: Will `asprintf()` help in that case?
Also things like `strlcpy()` are currently
available in glibc.
Just my 2 cents.

**Pull request merged**